src/java.desktop/macosx/classes/sun/lwawt/macosx/CTrayIcon.java

```diff
@@ -173,7 +173,7 @@ public void dispose() {
         }
 
         LWCToolkit.targetDisposedPeer(target, this);
-        target = null;
+//        target = null;
```

Comment:
Why do you commented out this line? just delete it and mark the target as final.

Reply:
you can also delete the null check(target == null) added by this fix:
https://cr.openjdk.org/~mhalder/8207938/webrev.01/src/java.desktop/macosx/classes/sun/lwawt/macosx/CTrayIcon.java.sdiff.html

BTW that bug looks like had the same root cause.